src/components/navigation/section-document-or-browser.tsx

```diff
@@ -146,22 +149,61 @@ export const SectionDocumentOrBrowser: React.FC<IProps> = observer(function Sect
 
     const sectionClass = openDocument?.type === "learningLog" ? "learning-log" : "";
 
+    const handleCollapseScroller = () => {
+      setScrollerCollapsed(!scrollerCollapsed);
+    };
     return (
-      <div>
-        <div className={`document-header ${tabSpec.tab} ${sectionClass}`} onClick={() => ui.setSelectedTile()}>
-          <div className={`document-title`}>
-            {getDocumentDisplayTitle(openDocument, appConfigStore, problemStore)}
+      <div className="scroller-and-document">
+        { selectedSubTab.label === "Starred" &&
+          <>
+            <div className={classNames("scroller", {"collapsed": scrollerCollapsed})} ref={documentScrollerRef}>
+              <div className={classNames("scroller-controls", "left", {"collapsed": scrollerCollapsed})}>
+                <div className="scroller-controls-overlay left"/>
+                <div className={classNames("scroll-arrow-button", "themed", tabSpec.tab,
+                                            {"collapsed": scrollerCollapsed}, {"show": showLeftScrollArrow})}>
+                  <ScrollArrowIcon className={`scroll-arrow left themed ${tabSpec.tab}`} />
+                </div>
+              </div>
```

Comment:
Would it be possible to wrap this into a component, then reuse it for lines 175-181 as well? I say this for two reasons. First, there are about a million `div`s in this render function, which makes it pretty difficult to understand. Second, it looks like there's a ton of overlap between this and 175-181, so it looks like a good opportunity to reduce duplicate code.